internal/controllers/reconciliation/controller.go

```diff
@@ -111,8 +115,23 @@ func (c *Controller) Reconcile(ctx context.Context, req *reconstitution.Request)
 		return ctrl.Result{}, nil
 	}
 
+	// Check if resource status is ready since we can't use the resource version cache for non-ready resources
+	// (otherwise we wouldn't be able to evaluate their readiness)
+	var ready bool
+	slice := &apiv1.ResourceSlice{}
+	err = c.client.Get(ctx, req.Manifest.Slice, slice)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("getting resource slice: %w", err)
+	}
+	if len(slice.Status.Resources) > req.Manifest.Index {
```

Comment:
Can the statuses be non-empty and have a different length than the manifests the slice holds?

Reply:
Practically, no. But it is technically possible if another client tampers with the status, so I think it's worth handling

Reply:
I fixed the logic and now the tests are failing lol. Will investigate in the morning